doc/nrf/libraries/networking/azure_iot_hub.rst

```diff
@@ -102,39 +123,170 @@ To connect to Azure IoT Hub using DPS, complete the following steps:
+
+The DPS APIs are documented in the :ref:`azure_iot_hub_dps_api` section.
+
+Connecting to Azure IoT Hub
+===========================
 
-Below is an example for setting the device ID at run time instead of compile time by configuring the :kconfig:option:`CONFIG_AZURE_IOT_HUB_DEVICE_ID` option:
+After the initialization, calling the :c:func:`azure_iot_hub_connect` function connects the device to the configured IoT hub or DPS instance, depending on the configuration.
+The initial TLS handshake takes a few seconds to complete, depending on the network conditions and the TLS cipher suite used.
+During the TLS handshake, the :c:func:`azure_iot_hub_connect` function blocks.
+Consider this when deciding the context from which the API is called.
+Optionally, DPS registration can be run automatically as part of the call to the :c:func:`azure_iot_hub_connect` function.
+Note that the :c:func:`azure_iot_hub_connect` function blocks when DPS registration is pending.
+Running DPS as part of the :c:func:`azure_iot_hub_connect` function also limits the DPS configuration options in several ways:
 
-   .. code-block:: c
+* The device ID is used as registration ID when registering with the DPS server.
+* The ID scope is set in the :kconfig:option:`CONFIG_AZURE_IOT_HUB_DPS_ID_SCOPE` option.
 
-	  struct azure_iot_hub_config cfg = {
-	      .device_id = "my-device",
-	      .device_id_len = sizeof("my-device") - 1,
-	  };
+Use the DPS APIs directly if you need more control over the DPS registration process.
 
-          int err = azure_iot_hub_init(&cfg, event_handler);
-          if (err) {
-              printk("azure_iot_hub_init failed: %d\n", err);
-	      return err;
-          }
+When using the :c:func:`azure_iot_hub_connect` function, you can choose to provide the hostname to the IoT Hub and device ID at run time, or let the library use Kconfig options.
 
-Connecting to Azure IoT Hub
-***************************
+Here is an example for setting the hostname and device ID at run time:
+
+.. code-block:: c
+
+   struct azure_iot_hub_config cfg = {
+      .hostname = {
+         .ptr = hostname_buffer,
+         .size = hostname_length,
+      },
+      .device_id = {
+         .ptr = device_id_buffer,
+         .size = device_id_length,
+      },
+      .use_dps = false,
+   };
 
-After the initialization, a :c:func:`azure_iot_hub_connect` function call connects the device to the configured IoT hub or DPS instance, depending on the configuration.
-The initial TLS handshake takes some time to complete, typically in the range of few seconds, depending on the network conditions and the TLS cipher suite used.
-During the TLS handshake, :c:func:`azure_iot_hub_connect` blocks, so care must be taken when deciding the context from which the API is called.
+   err = azure_iot_hub_connect(&cfg);
+   /* Error handling */
+
+You can pass ``NULL`` or a zeroed out configuration to the :c:func:`azure_iot_hub_connect` function.
```

Comment:
Not so sure about "zeroed out". How about an empty configuration? Or a configuration with no value?

Reply:
Cannot argue it is wrong. Maybe "cleared out" if you like, but we can leave it as it is. Up to you.